pkg/collector/corechecks/system/gpu/nvidia/jetson.go

```diff
@@ -0,0 +1,490 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+// +build !windows
+
+package nvidia
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/aggregator"
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+	core "github.com/DataDog/datadog-agent/pkg/collector/corechecks"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	checkName = "jetson"
+
+	kb = 1024
+	mb = kb * 1024
+	gb = mb * 1024
+
+	// Indices of the regex in the 'regexes' variable below
+	regexRAMIdx       = 0
+	regexSwapCacheIdx = 1
+	regexIRamIdx      = 2
+
+	// Regex used to parse the GPU usage and frequency => e.g. EMC_FREQ 7%@408 GR3D_FREQ 0%@76
+	regexGpuUsageIdx = 3
+
+	// Regex used to parse the CPU usage section => e.g. CPU [2%@102,1%@102,0%@102,0%@102]
+	regexCPUUsageIdx = 4
+
+	// Regex used to parse the temperature information => e.g. thermal@41C
+	regexTemperatureIdx = 5
+
+	// Regex used to parse the voltage information => e.g. POM_5V_IN 900/943
+	regexVoltageIdx = 6
+
+	// Regex used to parse cpu and freq => e.g. 2%@102
+	regexCPUFreqIdx = 7
+
+	// Indices of the matched fields by the RAM regex
```

Comment:
Since the regexp are hard coded. Could we have functions that return the information and encapsulate the logic for each instead of maintaining series of indexes ?

Instead of having:

```go
ramFields := c.regexes[regexRAMIdx].FindAllStringSubmatch(field, -1)
//[...]
ramMultiplier := getSizeMultiplier(ramFields[0][ramUnit])
```

I think this would be much easier to maintain:

```go
// either using a struct
ramDataStruct, err := getRamInfo(field)

// or even this if we want to avoid allocating a struct
ramMultiplier, ramUsed, ... := getRamInfo(field)
```